### PR TITLE
Test under JDK 18

### PR DESCRIPTION
**Pull request by mernst**
Merge together with
https://github.com/typetools/annotation-tools/pull/503
https://github.com/typetools/jdk/pull/146
https://github.com/typetools/jdk17u/pull/3

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -261,7 +263,7 @@ allprojects {
             options.deprecation = true
             options.compilerArgs += [
                     '-g',
-                    '-Werror',
+                    // '-Werror',
```
Comment (smillst) [inline build.gradle:266]: Should this change be reverted?

File: build.gradle [hunk 2]
```diff
@@ -310,7 +312,7 @@ allprojects {
                         // the errors get fixed.  On the downside, Error Prone (or maybe the compiler?)
                         // stops as soon as it issues one warning, rather than outputting them all.
                         // https://github.com/google/error-prone/issues/436
-                        '-Werror',
+                        // '-Werror',
```
Comment (smillst) [inline build.gradle:315]: Should this change be reverted?

File: checker/tests/fenum/UpperBoundsInByteCode.java [hunk 3]
```diff
@@ -1,3 +1,5 @@
+// @above-java17-jdk-skip-test TODO: reinstate on JDK 18, false positives may be due to issue #979
```
Comment (smillst) [inline checker/tests/fenum/UpperBoundsInByteCode.java:1]: Should I look into these warnings?

**Review by smillst — DISMISSED** [review]
I had two minor question, but otherwise looks good.

File: build.gradle (hunk 1 above)
Comment (mernst) [reply build.gradle:266]: Yes.  Thanks for catching this.

File: checker/tests/fenum/UpperBoundsInByteCode.java (hunk 3 above)
Comment (mernst) [reply checker/tests/fenum/UpperBoundsInByteCode.java:1]: Yes, please look into them as part of your work on #979, when you return to that task.

I don't have an opinion whether you should look into these problems first, or do other parts of the task that you know are essential before seeing whether these get fixed as a side effect.

**Pull request merged**